### PR TITLE
wsl: enable machine init tests

### PR DESCRIPTION
**Pull request by baude**
many of wsl's attributes are not as configurable as other providers. small tweaks and conditional tests have been added.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by n1hility** [comment]
LGTM

File: pkg/machine/e2e/init_test.go [hunk 1]
```diff
@@ -97,13 +98,16 @@ var _ = Describe("podman machine init", func() {
...
+		if testProvider.VMType() != machine.WSLVirt { // memory and cpus something we can set with WSL
+			Expect(testMachine.Resources.CPUs).To(Equal(uint64(cpus)))
+			Expect(testMachine.Resources.Memory).To(Equal(uint64(2048)))
+		}
 		Expect(testMachine.SSHConfig.RemoteUsername).To((Equal(remoteUsername)))
 
 	})
 
 	It("machine init with cpus, disk size, memory, timezone", func() {
+		skipIfWSL("setting hardware resource numbers and timezone are not supported on WSL")
```
Comment (vrothberg) [inline pkg/machine/e2e/init_test.go:110]: How does it behave on WSL?  Does it error out?  Any chance we can test against the expected behavior?

Comment (baude) [reply pkg/machine/e2e/init_test.go:110]: i have created #20175 for this, ok?

**Comment by mheon** [comment]
/lgtm
/hold
I suggest that we prioritize #20175 though

**Comment by TomSweeneyRedHat** [comment]
LGTM

File: pkg/machine/e2e/init_test.go (hunk 1 above)
Comment (vrothberg) [reply pkg/machine/e2e/init_test.go:110]: Thanks!

**Pull request merged**